src/frontend/lexer.rs

```diff
@@ -0,0 +1,259 @@
+use itertools::Itertools;
+use std::iter::Peekable;
+
+use super::token::Token;
```

Comment:
Per my understanding the hierarchy rust observes is Package > Crate > Module. Here I see that you have crates under your package, but no module (or in C++ speak "namespace"). I think its quite important to set something like a module hierarchy up to ensure that the scope of methods you create is controlled and stays readable. A simple idea here could be to start with something matching the directory layout like "mod lisprs::frontend::lexer{}"

Reply:
Check `src/frontend/mod.rs`. I'm pretty sure I do have the module tree mirroring the file tree.